users/user.go

```diff
@@ -0,0 +1,177 @@
+package users
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/LF-Engineering/dev-analytics-libraries/elastic"
+
+	"github.com/LF-Engineering/dev-analytics-libraries/auth0"
+	"github.com/LF-Engineering/dev-analytics-libraries/http"
+
+	json "github.com/json-iterator/go"
+)
+
+// Auth0ClientProvider ...
+type Auth0ClientProvider interface {
+	GetToken() (string, error)
+}
+
+// HTTPClientProvider ...
+type HTTPClientProvider interface {
+	Request(string, string, map[string]string, []byte, map[string]string) (int, []byte, error)
+}
+
+// ESClientProvider ...
+type ESClientProvider interface {
+	CreateDocument(index, documentID string, body []byte) ([]byte, error)
+	Search(index string, query map[string]interface{}) ([]byte, error)
+	CreateIndex(index string, body []byte) ([]byte, error)
+	Get(index string, query map[string]interface{}, result interface{}) error
+}
+
+// SlackProvider ...
+type SlackProvider interface {
+	SendText(text string) error
+}
+
+// Usr struct
+type Usr struct {
```

Comment:
Since this struct is the client functionalities receiver , so it's better to rename it to Client instead of Usr

Reply:
Will do

Reply:
Done